client/src/main/java/org/eclipse/hono/client/impl/CommandClientImpl.java

```diff
@@ -152,6 +161,49 @@ protected BufferResult getResult(final int status, final String contentType, fin
         });
     }
 
+    @Override
+    public Future<Void> sendOneWayCommand(final String command, final Buffer data) {
+        return sendOneWayCommand(command, null, data, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Future<Void> sendOneWayCommand(final String command, final String contentType, final Buffer data, final Map<String, Object> properties) {
+        Objects.requireNonNull(command);
+
+        //final Future<Void> responseTracker = Future.future();
+
+        final Future<BufferResult> responseTracker = Future.future();
+        final Span currentSpan = newChildSpan(null, command);
+
+        Optional.ofNullable(sender).ifPresent(s -> {
+            if (s.isOpen()) {
+                final Message request = ProtonHelper.message();
+
+                // set a unique correlationId for one-way requests, since there is no unique reply-id
+                request.setCorrelationId(UUID.randomUUID().toString());
```

Comment:
why do we need to set a correlation ID? There is no response expected so there will be nothing to correlate, right?

Reply:
No response, this is right - but to have a correlationId on the receiver (i.e. the device) should be reasonable, right?
And the here set correlationId would be the siginificant part of the `commandId`, that is sent to the device.

One could argue, though, that a one-way command does not need a commandId at all (to keep the footprint as low as possible).
Was that your concern?

Reply:
> One could argue, though, that a one-way command does not need a commandId at all (to keep the footprint as low as possible). Was that your concern?

yes, I do not see what it would be good for.

Reply:
Ok, we then handle one-way commands exactly the same as request-response commands - and only omit the `command-request-id`.

I will see what this will mean for MQTT (regarding the topic structure), but it seems be one valid way to go, which additionally has some advantages (low footprint).

Reply:
Maybe the MQTT adapter can simply use an empty string in that case, e.g. publishing to a topic like
`control///req//setBrightness`
analogous to the way we omit the tenant and device id ...

Reply:
Yes, this was my first though as well.
Looks good, probably the way to implement it for MQTT.